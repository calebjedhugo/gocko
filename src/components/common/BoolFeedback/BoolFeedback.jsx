import React from "react";
import Grow from "@material-ui/core/Grow";
import Paper from "@material-ui/core/Paper";
import Box from "@material-ui/core/Box";
import Typography from "@material-ui/core/Typography";
import PropTypes from "prop-types";

const BoolFeedback = ({
    visible = false,
    success = "Correct!",
    fail = "Wrong Answer. :(",
    value,
}) => (
    <Grow in={visible}>
        <Paper>
            <Box color={value ? "green" : "red"} textAlign="center" padding={1}>
                <Typography variant="h2">{value ? success : fail}</Typography>
            </Box>
        </Paper>
    </Grow>
);

BoolFeedback.propTypes = {
    visible: PropTypes.bool,
    success: PropTypes.string,
    fail: PropTypes.string,
    value: PropTypes.bool,
};

export default BoolFeedback;
