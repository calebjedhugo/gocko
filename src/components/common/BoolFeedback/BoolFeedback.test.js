import { render } from "@testing-library/react";
import BoolFeedback from "./BoolFeedback";

const defaultProps = { visible: true, value: true };

describe("<Header>", () => {
    it("Matches previous snapshot when visible and answer is correct", () =>
        expect(
            render(<BoolFeedback {...defaultProps} />).baseElement
        ).toMatchSnapshot());
    it("Matches previous snapshot when visible and answer is wrong", () =>
        expect(
            render(<BoolFeedback {...defaultProps} value={false} />).baseElement
        ).toMatchSnapshot());
    it("Matches previous snapshot when not visible", () =>
        expect(
            render(<BoolFeedback {...defaultProps} visible={false} />)
                .baseElement
        ).toMatchSnapshot());
});
