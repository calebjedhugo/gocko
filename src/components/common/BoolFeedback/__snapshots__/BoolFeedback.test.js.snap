// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header> Matches previous snapshot when not visible 1`] = `
<body>
  <div>
    <div
      class="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
      style="opacity: 0; transform: scale(0.75, 0.5625); visibility: hidden;"
    >
      <div
        class="MuiBox-root MuiBox-root-3"
      >
        <h2
          class="MuiTypography-root MuiTypography-h2"
        >
          Correct!
        </h2>
      </div>
    </div>
  </div>
</body>
`;

exports[`<Header> Matches previous snapshot when visible and answer is correct 1`] = `
<body>
  <div>
    <div
      class="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
      style="opacity: 1; transform: scale(1, 1); transition: opacity 0ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,transform 0ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    >
      <div
        class="MuiBox-root MuiBox-root-1"
      >
        <h2
          class="MuiTypography-root MuiTypography-h2"
        >
          Correct!
        </h2>
      </div>
    </div>
  </div>
</body>
`;

exports[`<Header> Matches previous snapshot when visible and answer is wrong 1`] = `
<body>
  <div>
    <div
      class="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
      style="opacity: 1; transform: scale(1, 1); transition: opacity 0ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,transform 0ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    >
      <div
        class="MuiBox-root MuiBox-root-2"
      >
        <h2
          class="MuiTypography-root MuiTypography-h2"
        >
          Wrong Answer. :(
        </h2>
      </div>
    </div>
  </div>
</body>
`;
